learning_table_tennis_from_scratch/rl_config.py

```diff
@@ -26,55 +26,76 @@ class PPOConfig:
         "n_epochs",
     )
 
+    _sac_params = (
+        "gamma",
+        "ent_coef",
+        "learning_rate",
+        "batch_size",
+        "gradient_steps",
+        "train_freq",
+        "buffer_size",
+        "learning_starts",
+    )
+
     # Additional parameters
     _additional_params = (
-        "num_timesteps",  # 'total_timesteps' passed to PPO.learn()
+        "num_timesteps",  # 'total_timesteps' passed to RL.learn()
         "save_path",  # If set the model is saved to the given path.
         "num_layers",  # Number of layers in the network
         "num_hidden",  # Size of each layer (all layers have same size)
         "log_path",  # Destination for checkpoints and log files
     )
 
-    __slots__ = _ppo_params + _additional_params
+    __slots_ppo__ = _ppo_params + _additional_params
+    __slots_sac__ = _sac_params + _additional_params
 
-    def __init__(self):
+    def __init__(self, algorithm):
+        if algorithm == "ppo":
+            self.__slots__ = self.__slots_ppo__
+            self.__algo_slots__ = self._ppo_params
+        elif algorithm == "sac":
+            self.__slots__ = self.__slots_sac__
+            self.__algo_slots__ = self._sac_params
```

Comment:
`__slots__` has a special meaning, it restricts the possible attributes that an object can have to the ones listed in `__slots__`.  However, this only works when it is defined at the class level.  By defining it here in the `__init__` method, it just becomes a normal variable without this special behaviour.  I think for the way how we use the class, this doesn't really matter, but it would probably be better to rename to something like `_params` it in this case, to avoid confusion.

Alternatively, I would suggest to change the architecture a bit to have a base class that defines all the methods and from this derive two classes `PPOConfig` and `SACConfig` which define `__slots__` (`from_dict`/`from_json` could then select the proper class based on the `algorithm` argument).

I think both solutions are okay, so go with the one you like more.